pkg/kubelet/volumes.go

```diff
@@ -87,11 +91,22 @@ func (vh *volumeHost) NewWrapperUnmounter(volName string, spec volume.Spec, podU
 		// Not found but not an error
 		return nil, nil
 	}
+
 	c, err := plugin.NewUnmounter(spec.Name(), podUID)
-	if err == nil && c == nil {
-		return nil, errUnsupportedVolumeType
+	if err != nil {
+		return nil, err
+	}
+	if c != nil {
+		// return good volume unmounter
+		return c, nil
 	}
-	return c, nil
+
+	// generic volume type error
+	return nil, createVolumeTypeError(volName)
```

Comment:
I have same concern as I articulated above.  `NewUnmounter` should never, ever return (nil, nil).